### DISCUSSION TITLE
Holidays on alternative calendars

### DISCUSSION THREAD

**Discussion by barakplasma** [azuyalabs/yasumi#293] [Ideas]
There are a few countries that use [calendars other than the gregorian calendar](https://en.wikipedia.org/wiki/Calendar#Usage:~:text=Chinese%2C%20Hebrew%2C%20Hindu%2C%20and%20Julian%20calendars%20are%20widely%20used%20for%20religious%20and%20social%20purposes.) (China, [Israel](https://en.wikipedia.org/wiki/Public_holidays_in_Israel#:~:text=Because%20of%20this%2C%20most%20holidays%20in%20Israel%20fall%20on%20a%20different%20Gregorian%20calendar%20date%20each%20year%2C%20which%20syncs%20every%2019%20years%20with%20the%20Hebrew%20calendar.), Hindu, etc), meaning the dates are different each year. Is this something yasumi wants to handle? Or would a static list of holidays be ok (for next 10-20 years)?

**Comment by stelgenhof** [comment]
Thank you for your question. The goal for Yasumi is to handle all holidays for all countries in a calculated way. There are indeed some countries of which the holidays are perhaps hard to define dynamically. As such, the aim is to use any algorithm or pattern that may exist and use that. Using a static list would be a last resort in my view.

**Reply by barakplasma** [reply]
Thanks for the reply.

There's an existing C, JS, and Go library for calculating the holidays of the Hebrew calendar ( [hebcal/hebcal](https://github.com/hebcal/hebcal) ). Is integrating an external library via Foreign Function Interface a good option?

**Reply by stelgenhof** [reply]
Although interesting, it would add a (non-PHP) dependency. The idea of Yasumi is to be a PHP only library to allow for easy installation and use. Wouldn't it be an idea to port the logic from that library to PHP?
